Implement suspendable method for the ISuspendable interface

matthias-wende-sociomantic opened the pull request:
Fixes https://github.com/sociomantic-tsunami/swarm/issues/391

matthias-wende-sociomantic committed:
Add suspended method to Controller implementation

... of BatchRequestCore and SuspendableRequestCore.

This adds the possibility to add a full working implementation
of oceans ISuspendable interface in client requests using
the BatchRequestCore.

Note that the method is actually only implementable in the
BatchRequestCore and will throw in the SuspendableRequestCore.
The proper fix is to replace the latter by the former.

matthias-wende-sociomantic committed:
Use new Controller suspended method in Controllers

This implements the `suspended` method in the Controllers
ISuspendable implementation using the `suspended` method
recently introduced in BatchRequestCore and SuspendableRequestCore.

gavin-norman-sociomantic commented on src/swarm/neo/client/mixins/Controllers.d:
```diff
@@ -303,25 +311,23 @@ template Controllers ( )
+                }
+            );
+
+            return this._suspended;
```
Does this need to be a class member? Could it not just be a local?

gavin-norman-sociomantic commented on src/swarm/neo/client/mixins/Controllers.d:
```diff
@@ -303,25 +311,23 @@ template Controllers ( )
-            being controlled, in order to receive the suspended/resumed
-            notifications to know when the request had actually changed state
-            (on the nodes' side).
+            TODO add description
```
What did you have in mind here? I'm not sure if we need to say anything beyond what Returns states now.

matthias-wende-sociomantic committed:
Deprecate handlePending method

The handlePending() was used to initiate a state-change. These state
changes are used in requests using the SuspendableRequestCore which is now
also deprecated and superseded by the BatchRequestCore.

gavin-norman-sociomantic commented on src/swarm/neo/client/mixins/Controllers.d:
```diff
@@ -281,6 +281,7 @@ template Controllers ( )
 
         ***********************************************************************/
 
+        deprecated("Only works with SuspendableRequestCore which is deprecated")
```
Ah sorry, this will need an accompanying deprecation release note.

matthias-wende-sociomantic commented:
Rebased and added release nodes.
